packages/store/src/internal/state-operations.ts

```diff
@@ -12,11 +12,19 @@ import { deepFreeze } from '../utils/freeze';
  */
 @Injectable()
 export class InternalStateOperations {
-  constructor(
-    private _stateStream: StateStream,
-    private _dispatcher: InternalDispatcher,
-    private _config: NgxsConfig
-  ) {}
+  constructor(private _stateStream: StateStream, private _dispatcher: InternalDispatcher, private _config: NgxsConfig) {
+    InternalStateOperations.checkDevelopmentMode(_config.developmentMode);
+  }
+
+  private static checkDevelopmentMode(developmentMode: boolean) {
```

Comment:
This doesn't need to be static. Just call the instance method and this method could check the `this._config.developmentMode` value.